lib/install/actions.js

```diff
@@ -46,7 +46,8 @@ Object.keys(actions).forEach(function (actionName) {
       if (!pkg.commit) pkg.commit = []
       pkg.commit.push(action.commit)
     }
-    if (pkg.knownInstallable) {
+
+    if (pkg.knownInstallable || pkg.package._optional) {
```

Comment:
True. It should be handled through the errorMessage like the general `optionalDependencies` way of `install.js`. I will update them again!